### PR TITLE
create RBAC for accessing claimed cluster

### PR DESCRIPTION
**Pull request by staebler**
When a ClusterClaim is assigned a cluster, the clusterclaim controller will create a Role and RoleBinding in the namespace
of the ClusterDeployment. The Role will grant full permissions for Hive resources and GET permissions for the admin kubeconfig and password secrets. The RoleBinding will assign the role to the subjects specified in the ClusterClaim.

https://issues.redhat.com/browse/CO-1070

### CODE REVIEW

**Comment by staebler** [comment]
This builds on #1065.

/hold

File: pkg/apis/hive/v1/clusterclaim_types.go [hunk 1]
```diff
@@ -2,16 +2,17 @@ package v1
...
 
 // ClusterClaimSpec defines the desired state of the ClusterClaim.
 type ClusterClaimSpec struct {
 	ClusterPoolName string `json:"clusterPoolName"`
 
-	// Subjects hold object references to authorize access to the claimed cluster.
+	// Subjects hold references to which to authorize access to the claimed cluster.
 	// +optional
-	Subjects []corev1.ObjectReference `json:"subjects,omitempty"`
+	Subjects []rbacv1.Subject `json:"subjects,omitempty"`
```
Comment (dgoodwin) [inline pkg/apis/hive/v1/clusterclaim_types.go:15]: How far have you tested this? I'd imagine we're most likely talking about OpenShift users here, but that concept does not exist in Kube IIRC. Will these Subjects be able to hold and work with users?

Comment (staebler) [reply pkg/apis/hive/v1/clusterclaim_types.go:15]: Kube does have a concept of users. As far as I understand it, an OpenShift user is a Kube user. A subject can refer to a group, a user, or a service account.

**Comment by dgoodwin** [comment]
Needs rebase but looks good, will take another look as soon as #1065 goes in.

**Comment by staebler** [comment]
/hold cancel

#1065 has merged.

File: pkg/controller/clusterclaim/clusterclaim_controller.go [hunk 2]
```diff
@@ -48,14 +56,87 @@ func AddToManager(mgr manager.Manager, r *ReconcileClusterClaim) error {
...
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: requestsForRBACResources(r.Client, hiveClaimOwnerRoleName, r.logger),
+		},
+	); err != nil {
+		return err
+	}
+
+	// Watch for changes to the hive-claim-owner RoleBinding
+	if err := c.Watch(
+		&source.Kind{Type: &rbacv1.Role{}},
```
Comment (2uasimojo) [inline pkg/controller/clusterclaim/clusterclaim_controller.go:85]:
```suggestion
		&source.Kind{Type: &rbacv1.RoleBinding{}},
```

copypasta? See #2024

**Pull request merged**